tests/pytest.ini

```diff
@@ -33,3 +33,6 @@ filterwarnings =
     ignore:PauliWord.hamiltonian:pennylane.PennyLaneDeprecationWarning
 addopts = --benchmark-disable
 xfail_strict=true
+
+[tool:pytest]
+rng_salt = v0.38.0
```

Comment:
So this salt determines the seed that gets passed to all the tests? How does that work?
* Does the salt get used to generate a random integer that is then used as the seed for all tests?
* Or, does the salt get used to generate a seed randomly for each test that uses the seed fixture?
  * If it's the latter, is `pytest-rng` thread-safe, i.e., would a given test always get the same seed regardless of which tests run before/after it?
* If changing the salt value will affect the seeds such that the new seed(s) cause tests to fail, that would still require us to manually address this, right? If this is the case, then it definitely makes sense to merge this into `master` after the RC branch is created.

Reply:
- The salt is used by `pytest-rng` to generate a different random integer for each test, but it guarantees that the same test always gets the same seed.
- I took a look at the source code for `pytest-rng`, and it turns out that the seed it gives to each test is dependent only on the `nodeid` of the test (created by pytest). This `nodeid` is unique and it does not change regardless of how or when the test is run.
- Yes, when we change the salt value, we will need to manually address the failing tests, i.e., create bug reports for actual bugs, or augment the seed locally to make the test pass if the failure is harmless, which is precisely the point of using `pytest-rng`: to periodically check all seeded stochastic tests to make sure that they're not passing because of a magic seed that happen to work.
- In this PR I will update all seeded tests to use the `pytest-rng` generated seed, and by the time this PR is complete, all of these tests will always pass with the current salt that is set. We can safely merge it into master this week. As long as we don't change the salt on the RC branch, all tests will continue to pass.

Reply:
The `nodeid` and thus the seed given to a particular test case will change if you change the name of the test or move the test to a different location, but it will remain the same as you modify, add, or remove other tests across the test suite, so I say it should be pretty stable.

Reply:
Thanks, that's really helpful context. From what you've said, I don't see any issues with merging this this week as long as:
1. We don't change the salt value until after all release-related tasks are completed.
2. We don't change test names.

The 2nd point is not something that I feel like we can guarantee, and it might not make sense to enforce it either in case there are typos or other reasons to change a test's name. Let's use today's code discussion in stand-up to finalize the decision.

Reply:
If you change the name of a test and the test fails because of it, all you need to do is to augment the seed locally to make it pass again. This won't affect anyone else. Although this is indeed something that needs to be communicated to the team.